crates/core/src/db/datastore/traits.rs

```diff
@@ -194,6 +196,22 @@ impl TableSchema {
     pub fn normalize_field(&self, or_use: &TableField) -> FieldName {
         FieldName::named(or_use.table.unwrap_or(&self.table_name), or_use.field)
     }
+
+    /// Project the fields from the supplied `indexes`.
+    pub fn project(&self, indexes: impl Iterator<Item = usize>) -> Result<Vec<&ColumnSchema>> {
+        indexes
+            .map(|index| {
+                self.get_column(index)
+                    .ok_or(InvalidFieldError { index, name: None }.into())
+            })
+            .collect()
+    }
+
+    /// Utility for project the fields from the supplied `indexes` that is a [NonEmpty<u32>],
+    /// used for when the list of field indexes have at least one value.
+    pub fn project_not_empty(&self, indexes: &NonEmpty<u32>) -> Result<Vec<&ColumnSchema>> {
```

Comment:
Calling the argument `indexes` makes no sense here. You're projecting a columns by their column `id`. It's actually unrelated to indexes. It should be called `col_ids` or `cols` or something like that.